test/git_repository.go

```diff
@@ -72,6 +72,19 @@ func NewGitRepository(workingDir string, homeDir string, shell command.Shell) Gi
 	return GitRepository{Dir: workingDir, Shell: shell}
 }
 
+// HasBranchesOutOfSync returns if true if and only if one or more local branches are out of sync with their remote
+func (repo *GitRepository) HasBranchesOutOfSync() (bool, error) {
+	res, err := repo.Shell.Run("bash", "-c", "git branch -vv | grep -o \"\\[.*\\]\" | tr -d \"[]\" | awk \"{ print \\$2 }\" | grep . | wc -l")
+	if err != nil {
+		return false, fmt.Errorf("cannot determine if any branches are out of sync in %q: %w %q", repo.Dir, err, res.Output())
+	}
+	count, err := strconv.ParseInt(res.OutputSanitized(), 0, 64)
```

Comment:
Note went back to parsing count as grep returns exit code 1 if no rows are found